pallets/stake/src/lib.rs

```diff
@@ -14,7 +14,30 @@
 // You should have received a copy of the GNU General Public License
 // along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Minimal staking module with ordered validator selection
+//! # Stake
+//! Minimal staking pallet that implements ordered validator selection by total amount at stake
+//!
+//! ### Rules
+//! There is a new round every `BlocksPerRound` blocks.
+//!
+//! At the start of every round,
+//! * `IssuancePerRound` is distributed to validators for `BondDuration` rounds ago
+//! in proportion to the points they received in that round (for authoring blocks)
+//! * queued validator exits are executed
+//! * new set of validators is chosen from the candidates
+//!
+//! To join the set of candidates, an account must call `join_candidates` with
+//! stake >= `MinValidatorStk` and fee <= `MaxFee`. The fee is taken off the top
+//! of any rewards for the validator before the remaining capital is distributed to nominators in
+//! proportion to stake (and the validator always self-nominates)
+//!
+//! To leave the set of candidates, the validator calls `leave_candidates` and is immediately
+//! removed from candidates and queued in `ExitQueue` to be removed `BondDuration` rounds later.
```

Comment:
I don't understand, it is removed immediately or after `BondDuration` rounds later ?
Also, is it removed from the current validators if it is in ?

Reply:
The validator is immediately removed from the `CandidateQueue` so they will not be selected in future validator sets. 

They are not removed from the current validator set because they only made the request then and they should have some retroactive accountability (at some point) so bonding them for `BondDuration` rounds and only executing the exit after those rounds is safest.

The current validator set is overwritten before the start of every round. It is overwritten with the top `MaxValidators` from the `CandidateQueue`, ordered by stake. So, when a validator requests exit, we remove them from this pool of candidates so they cannot be selected for future validator sets, but we do not unstake them until `BondDuration` rounds later. We queue the request to leave in the `ExitQueue` storage item.

Reply:
Sounds good, maybe adding those clarification in the code comments would be good